osbs/build/spec.py

```diff
@@ -334,6 +324,29 @@ def set_params(self, git_uri=None, git_ref=None,
         self.koji_upload_dir.value = koji_upload_dir
         self.yum_proxy.value = yum_proxy
 
+        self._populate_image_tag()
+
+    def _populate_image_tag(self):
+        timestamp = utcnow().strftime('%Y%m%d%H%M%S')
+        # RNG is seeded once its imported, so in cli calls scratch builds would get unique name.
+        # On brew builders we import osbs once - thus RNG is seeded once and `randrange`
+        # returns the same values throughout the life of the builder.
+        # Before each `randrange` call we should be calling `.seed` to prevent this
+        random.seed()
+
+        tag_segments = [
+            self.koji_target.value or 'none',
+            str(random.randrange(10**(RAND_DIGITS - 1), 10**RAND_DIGITS)),
+            timestamp
+        ]
+
+        # Support for platform specific tags has only been added in arrangement 3.
+        if self.platform.value and (self.arrangement_version.value or 0) >= 3:
```

Comment:
After discussion today I think this will need to wait for arrangement 4 -- agree?